Sources/FoundationExtensions/LossyCollections.swift

```diff
@@ -219,3 +219,64 @@ extension LossyArrayDictionary: ExpressibleByDictionaryLiteral {
     }
 
 }
+
+// MARK: -
+
+// From https://github.com/apple/swift-corelibs-foundation/blob/swift-5.6-RELEASE/Sources/Foundation/JSONEncoder.swift
+private func convertToSnakeCase(_ stringKey: String) -> String {
```

Comment:
Unfortunately this isn't working well in all cases with a combination of identifiers like
- `com.revenuecat.monthly_4.99.no_intro`
- `com.revenuecat.monthly_4.99.1_free_week`
- `pro.1`

Those get converted to camelCase, and converting them back to snake case can't be done consistently.

Reply:
I'll have to find a workaround that allows `LossyDictionary` to not convert to camel case when decoding `DictionaryCodingKey`s.

Reply:
With the workaround:
> failed - expected to equal <[com.revenuecat.monthly_4.99.1_week_intro, com.revenuecat.monthly_4.99.no_intro, pro.1, snake_case]>, got <[com.revenuecat.monthly_4.99.1_week_intro, com.revenuecat.monthly_4.99._no_intro, pro._1, snake_case]>, missing <[com.revenuecat.monthly_4.99.no_intro, pro.1]>, extra <[com.revenuecat.monthly_4.99._no_intro, pro._1]>

Without the workaround:

> failed - expected to equal <[com.revenuecat.monthly_4.99.1_week_intro, com.revenuecat.monthly_4.99.no_intro, pro.1, snake_case]>, got <[com.revenuecat.monthly4.99.1WeekIntro, com.revenuecat.monthly4.99.NoIntro, pro.1, snakeCase]>, missing <[com.revenuecat.monthly_4.99.1_week_intro, com.revenuecat.monthly_4.99.no_intro, snake_case]>, extra <[com.revenuecat.monthly4.99.1WeekIntro, com.revenuecat.monthly4.99.NoIntro, snakeCase]>